controllers/rating_controller.go

```diff
@@ -82,21 +82,59 @@ func (r *RatingReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		}
 		return reconcile.Result{}, err
 	}
+	component := corev1alpha1.Component{}
+	if err := r.Client.Get(ctx, types.NamespacedName{Namespace: instance.Namespace, Name: instance.Spec.ComponentName}, &component); err != nil {
```

Comment:
If component or repository is not found, we should just return, do not reconcile again and again....

Reply:
Requeue tells the Controller to requeue the reconcile key.  Defaults to false.

Reply:
https://github.com/kubernetes-sigs/controller-runtime/blob/v0.12.3/pkg/internal/controller/controller.go#L323

If you return error, the req will also requeue...

A detailed description of this customary practice can be found in the example:
https://github.com/kubernetes-sigs/kubebuilder/blob/master/docs/book/src/cronjob-tutorial/testdata/project/internal/controller/cronjob_controller.go#L117